indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskStorageQueryAdapter.java

```diff
@@ -114,7 +114,7 @@ public Optional<Task> getTask(final String taskid)
           return activeTask;
         }
       }
-      catch (JsonProcessingException e) {
+      catch (DruidException e) {
```

Comment:
Why are we swallowing this exception?

Reply:
I thought it might be helpful to fallback to the metadata store (TaskStorage) in such a case

Reply:
We have to do one of two things:
Either (1) Throw `JsonProcessingException` from `TaskQueue.getActiveTask()`, catch it here, log it and fallback to using `TaskStorage` (catch block should have a comment to this effect). In this case, `TaskQueue.getActiveTask` need not log the exception.
Or (2) Throw `DruidException` from `TaskQueue.getActiveTask()` and don't catch it anywhere.

I would prefer the latter as exception during serde shouldn't happen, and it is best to notify the operator about it as soon as possible.
Also if there is an issue deserializing the task, the task storage would be likely to encounter the same issue as it is the same mapper and the same Task object.

Reply:
Done